src/pombot/commands/iwill.js

```diff
@@ -1,11 +1,27 @@
 /*
  * A stub for the "i will" command.
  */
-import {createCommand} from 'chatter';
 
-export default createCommand({
-  name: 'i will',
-  description: 'Records the users task for the upcoming pom.',
-}, () => {
-  return `> Theoretically we will all be doing things.`;
-});
+import {createCommand, createParser} from 'chatter';
+import {states} from '../pomConfig';
+
+export default function(pom) {
+  return createCommand({
+    name: 'i will',
+    description: 'Records a users task for the next pom.',
+  }, (message, {user, channel}) => {
+    switch (pom.state) {
+      case states.RUNNING:
+        return `🍅 it's too late to declare a task, a pom is running with *${pom.getTimeString(pom.timeLeft)}* left.`;
+        break;
+      default:
+        // if pom is not running or is on break, let user record task
+        return createParser(parsed => {
```

Comment:
Why create and return a new message handler here? Since you're only using `parsed.input`, just use `message`, since it's the same thing.